File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp;
        vector<int> ans;
        for (int i = 0; i < nums.size(); i++)
            mp[nums[i]] = i;
        for(int i = 0; i < nums.size(); i++){
            int sec = target - nums[i];
            if(mp.find(sec) != mp.end() && mp[sec] != i){
                ans.push_back(i), ans.push_back(mp[sec]);
                break;
            }
        }
        return ans;
    }
};

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        // a + b = sum
        int[] ans = new int[2];
        Map<Integer, Integer> mp = new HashMap<>();
        for(int i = 0; i < n; i++){
            int a = nums[i];
            if(mp.containsKey(target - a)){
                ans[0] = i;
                ans[1] = mp.get(target - a);
                return ans;
            }
            mp.put(a, i);
        }
        return ans;
    }
}